scripts/tools/tools-object.js

```diff
@@ -0,0 +1,57 @@
+const schema = require("./tools-schema.json");
+const axios = require('axios')
+var validate = require("jsonschema").validate;
+
+let appendData = {
+    generator: [],
+};
+
+const makeObject = (object, repositoryUrl, isAsyncAPIrepo) => {
+    let resultantObject = {
+      title: object.title,
+      description: object.description,
+      links: {
+        websiteUrl: object.links.websiteUrl,
+        docsUrl: object.links.docsUrl,
+        iconUrl: object.links.iconUrl,
+        repoUrl: repositoryUrl,
+      },
+      filters: {
+        languages: object.filters.language,
+        technology: object.filters.technology,
+        categories: object.filters.categories,
+        hasCommercial: object.filters.hasCommercial,
+        isAsyncAPIOwner: isAsyncAPIrepo,
+      },
+    };
+    return resultantObject;
+  };
+  
+  const getContent = async (result_object) => {
+    let reference_id = result_object.url.split("=")[1];
+    let download_url = `https://raw.githubusercontent.com/${result_object.repository.full_name}/${reference_id}/${result_object.path}`;
+    //   console.log(download_url);
+    let toolObject;
+    const resp = await axios.get(download_url);
+    if (!validate(resp.data, schema).errors.length) {
+      let repositoryUrl = result_object.html_url;
+      let isAsyncAPIrepo =
+        result_object.repository.owner.login === "asyncapi" ? true : false;
+      toolObject = makeObject(resp.data, repositoryUrl, isAsyncAPIrepo);
+      resp.data.filters.categories.forEach((category) => {
+        appendData[category].push(toolObject);
+      });
+    }
+};
+
+async function convertTools(data) {
+    const dataArray = data.items;
+    let len = dataArray.length;
+    for(let i =0;i<len;i++){
+      if(dataArray[i].name === '.asyncapi-tool')
+        await getContent(dataArray[i]);
+    }
+    return appendData;
```

Comment:
Ohh, don't worry, I actually wrote this return statement while implementing the Promises but as you can see, we are not using the Promises anymore, I will remove this line :sweat_smile:.

Reply:
I think we have to share the `appendData` object among functions because multiple tools can have many technologies. categories and languages. Making this variable global will allow us to access this in any function wherever it is needed and we don't have to then pass this object as parameter for each call of tool.

Regarding your point of getting some mismatched data, this can't be possible because we are running each computation here synchronously (not async) so, I don't think it will make errors in synchronous call of tools.

Reply:
> I think we have to share the appendData object among functions because multiple tools can have many technologies. categories and languages

why is it a blocker? only `getContent` operates on that object anyway, or?

> Regarding your point of getting some mismatched data

I'm not saying there will be mismatched data. It is more about the maintainability of the code. I was super confused to see how that works with `appendData`, and it can be error-prone long term.

Reply:
> why is it a blocker? only getContent operates on that object anyway, or?

`convertTools` function also use this variable to return the full object. See, we are calling `getContent` function for each tool, not for all tools at once. So if we defined the `appendData` inside that, we won't get that good results. If you want to have that variable locally defined in a function, we can have just one function, instead of decomposing it to one, this you want?

Reply:
yes, I know `convertTools` depends on it and this is why I suggest to improve `convertTools` by adding `appendData` there

Reply:
Fine, I will change the `convertTools` function accordingly.